tt_metal/tt_stl/reflection.hpp

```diff
@@ -1041,7 +1041,8 @@ inline hash_t hash_object(const std::variant<Ts...>& variant) noexcept {
     if constexpr (DEBUG_HASH_OBJECT_FUNCTION) {
         fmt::print("Hashing std::variant: {}\n", variant);
     }
-    return std::visit([](const auto& value) { return hash_object(value); }, variant);
+    auto active_variant = variant.index();
+    return std::visit([&](const auto& value) { return hash_objects(active_variant, value); }, variant);
```

Comment:
Could you elaborate why this change was introduced?

Reply:
I see. So you have 2 structures with different meanings, but their hashing otherwise is the same. Or say you have 2 different empty structures. So you must hash the type in. Thank you for the clarification.

Reply:
Gotcha, I don't have anything against this change. Out of curiosity, I tried this:

```rust
use std::hash::{DefaultHasher, Hash, Hasher};

#[derive(Hash)]
struct RowMajorPageConfig {
    x: u32
}

#[derive(Hash)]
struct TilePageConfig {
    x: u32
}

#[derive(Hash)]
enum Config {
    RowMajorPageConfig(RowMajorPageConfig),
    TilePageConfig(TilePageConfig)
}

fn calculate_hash<T: Hash>(t: &T) -> u64 {
    let mut s = DefaultHasher::new();
    t.hash(&mut s);
    s.finish()
}


pub fn main() {
    println!("{}", calculate_hash(&RowMajorPageConfig{x: 10}));
    println!("{}", calculate_hash(&TilePageConfig{x: 10}));
    println!("{}", calculate_hash(&Config::TilePageConfig(TilePageConfig{x: 10})));
}
```

and I got

```
7396300056459196583
7396300056459196583
17156080100134691638
```

So yeah, Rust hashes an enum with the value differently from the value itself, so we're good!

Specifically in regards to `RowMajorPageConfig`, there are no member variables in that class, so I think what you get is just the hash of the default seed value. So it would be the same for any other empty class. All I'm saying is that this fix might not be enough to correctly handle such situations. It could be better to create a custom hash method.